tcping.go

```diff
@@ -578,16 +585,16 @@ func nanoToMillisecond(nano int64) float32 {
 }
 
 // handleConnError processes failed probes
-func (tcpStats *stats) handleConnError(connTime time.Time) {
+func (tcpStats *stats) handleConnError(connTime time.Time, elapsed time.Duration) {
 	if !tcpStats.wasDown {
 		tcpStats.startOfDowntime = connTime
-		calcLongestUptime(tcpStats,
-			time.Duration(tcpStats.ongoingSuccessfulProbes)*time.Second)
+		uptime := time.Since(tcpStats.startOfUptime).Truncate(time.Second)
```

Comment:
I think `Truncate` is causing some very tiny "miscalculations".

Check out the `longest consecutive uptime` value in these two images:

![image](https://github.com/pouriyajamshidi/tcping/assets/54482226/2f1d1d25-a967-49aa-aeaf-d7d3209c975f)


And then:


![image](https://github.com/pouriyajamshidi/tcping/assets/54482226/238e604b-cb4d-4d9c-954f-fbac3a5e268c)


You have to run it a few times to reproduce but my guess is, if the connection duration falls under a certain value, it just truncates it.